packages/authorization-process/src/services/validators.ts

```diff
@@ -125,3 +127,13 @@ export const assertKeyDoesNotAlreadyExist = async (
     throw keyAlreadyExists(kid);
   }
 };
+
+export function assertValidateRsaKeyLength(
+  jwk: JsonWebKey,
+  minLength: number = 2048
+): void {
+  const keyLength = jwk.n && Buffer.from(jwk.n, "base64url").length * 8;
+  if (!keyLength || (keyLength && keyLength < minLength)) {
+    throw invalidKeyLength(JSON.stringify(jwk));
+  }
+}
```

Comment:
Instead of doing this from the JWK, it's easier and more readable to check the length of the `KeyObject` we create before exporting the JWK. We could:
1. refactor the assertion as follows ⬇️ 
2. call it inside `createPublicKey` before returning the `KeyObject`

In this way every public key we create gets checked in length (createPublicKey is called in `createJWK` so the check will happen in both the functions you modified.

```suggestion
function assertValidRSAKeyLength(
  key: KeyObject,
  minLength: number = 2048
): void {
  const length = key.asymmetricKeyDetails?.modulusLength;
  if (!length || length < minLength) {
    throw invalidKeyLength(length, minLength);
  }
}